### ISSUE TITLE
Override http.Client or http.Transport in use by SDK?

### ISSUE THREAD

**Issue by ncw** [Azure/azure-storage-blob-go#79]
Is it possible to override either the `http.Client` or `http.Transport` in use by the SDK?

This is very important for restricted environments, eg [on google AppEngine](https://godoc.org/google.golang.org/appengine/urlfetch).

Rclone uses this functionality to add http debugging and idle timeouts.

I'll just note that the AWS SDK and the Google SDK both allow overriding it :-)

Thanks

**Comment by JeffreyRichter** [comment]
Yes, it is possible to do. We do consider it a more advanced scenario. You can see an example of us doing it with our AzCopy utility [here](https://github.com/Azure/azure-storage-azcopy/blob/f967afe7fdf05b42e376abafff4d65bee295514a/ste/mgr-JobPartMgr.go#L65). 

Basically, you have to call pipeline package's NewPipeline function passing in pipeline.Options after setting the HTTPSender field to the desired httpClient/transport you desire. NewPipeline also requires a slice of pipeline.Factory objects (which are also completely customizable but considered an advanced scenario) to manually configure pipeline policies such as telemetry, request IDs, credentials, retries, logging, etc. azblob's NewPipeline [function](https://github.com/Azure/azure-storage-blob-go/blob/67440baa31716bc29c9d0c48b1653f9afad3909b/2017-07-29/azblob/zc_pipeline.go) does this for you to address the common user case. You do NOT have to call azblob's NewPipeline to create a pipeline, you can manually create a pipeline to have more control over it.

**Comment by ncw** [comment]
Thank you for your explanation - much appreciated :-)
